corehq/apps/reports/filters/users.py

```diff
@@ -315,13 +316,18 @@ def user_es_query(cls, domain, mobile_user_and_group_slugs):
     @classmethod
     def pull_users_and_groups(cls, domain, mobile_user_and_group_slugs,
                               include_inactive=False, limit_user_ids=None):
-        user_ids = cls.selected_user_ids(mobile_user_and_group_slugs)
+        user_ids = set(cls.selected_user_ids(mobile_user_and_group_slugs))
         user_types = cls.selected_user_types(mobile_user_and_group_slugs)
         group_ids = cls.selected_group_ids(mobile_user_and_group_slugs)
+        location_ids = cls.selected_location_ids(mobile_user_and_group_slugs)
         users = []
 
         if limit_user_ids:
-            user_ids = set(limit_user_ids).intersection(set(user_ids))
+            user_ids = set(limit_user_ids).intersection(user_ids)
```

Comment:
https://github.com/dimagi/commcare-hq/pull/14397/commits/37b3c21d53ba02b2014fcd523c8f214e19198ccd -- this function could use some tests. going to leave for now so we can patch up icds, but will followup with a test